### ISSUE TITLE
sys.argv[0] Incorrect in onefile windows builds.

### ISSUE THREAD

**Issue by DevSean** [Nuitka/Nuitka#907]
### Information
Version:

nuitka --version
0.6.10.1
Python: 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)]
Executable: C:\Users\Sean\AppData\Local\Programs\Python\Python38\python.exe
OS: Windows
Arch: x86_64

I Installed Nuitka using pip

### Problem
sys.argv[0] returns:
> C:\Users\Sean\AppData\Local\example\example\1.0.0.0\example.exe

Expected:
> D:\Git\sys.argv-issue\example.py

Example:

```
import os
import sys

filePath = os.path.join(os.path.dirname(sys.argv[0]), "test-file.txt")

outputFile = open(filePath, "w+", encoding="utf-8")
outputFile.writelines(filePath)
print("Wrote file to: " + filePath)
outputFile.close()

inputFile = open(filePath, "r", encoding="utf-8")
print("Read file from: " + inputFile.readline())
inputFile.close()

while True:
    pass
```

The output from a standalone using build command:
`nuitka example.py --mingw64 --standalone`

> Wrote file to: D:\Git\sys.argv-issue\example.dist\test-file.txt
> Read file from: D:\Git\sys.argv-issue\example.dist\test-file.txt

Output from a onefile using build command:
`nuitka example.py --mingw64 --standalone --onefile --windows-product-version=1.0.0 --windows-company-name=example`

> Wrote file to: C:\Users\Sean\AppData\Local\example\example\1.0.0.0\test-file.txt
> Read file from: C:\Users\Sean\AppData\Local\example\example\1.0.0.0\test-file.txt

Note, the following alternatives had the same problem:
```
filePath = os.path.join(sys.path[0], "test-file.txt")
filePath = os.path.join(os.path.dirname(os.path.realpath(__file__)), "test-file.txt")
```

I am able to get the directory using the following but not the executable path:
`filePath = os.path.join(os.path.realpath(""), "test-file.txt")`

**Comment by kayhayen** [comment]
It works as expected, the onefile binary bootstraps the files it unpacked to the local appdata folder and that is the only file ever relevant. Granted, onefile currently doesn't yet allow to include data files, something I want to change soon, but they would live aside that executable.

You may find it as the process current directory, but currently it's not foreseen to put it in `sys.argv`, definitely not in `__file__`, that app folder is effectively the `.dist` folder installed. I could image providing the path to the launching binary if any in an environment variable, but there isn*t really any good way to tell it. Maybe if I introduce a secret command line option like multiprocessing module does.

Can you elabortate a bit more on your use case, why do you want to work with files near the onefile binary. Is that for your program output?

**Comment by DevSean** [comment]
I wanted to be able to read/write files in a human-readable format alongside the executable.

I wasn't aware the onefile binary would bootstrap to the AppData local folder, the behaviour I experienced makes sense now.

I encountered this problem when I attempted to load some pre-made files that I had placed alongside the executable. My program silently failed to read the files; it took me a little bit of debugging to understand why.

This behaviour is not blocking me in any way, as such; I am happy for this issue to be closed if you prefer.

**Comment by kayhayen** [comment]
I was wondering how to deal with this, there is a conflict with how `__file__` is typically used. I am thinking now, that we may in fact change `sys.argv[0]` to retain the original path, if it was launched through the onefile, but then there is no way to ever find the binary that makes most sense to call.

So I am thinking of adding `__onefile__`  for non-Windows to use and to be the bootstrap binary, but that leads to incompatible code with CPython. Maybe adding it to `__compiled__` as an extra element could be used, where we would also have the actual binary path with `.exe` if applicable, so that is more clean.

**Comment by kayhayen** [comment]
So `sys.argv[0]` is correct now, it's the bootstrap binary if that is what is called, and otherwise the unpacked binary if that is what is called by the user. There is also a new mode to put it into a temporary folder, that works that way too.

**Comment by kayhayen** [comment]
Released as part of 0.6.11 just now.
